Explorer/Assets/DCL/SDKComponents/TriggerArea/Systems/TriggerAreaHandlerSystem.cs

```diff
@@ -0,0 +1,225 @@
+            ProcessOnStayInTriggerArea(entity, triggerAreaCRDTEntity, transform, ref triggerAreaComponent);
+            ProcessOnExitTriggerArea(entity, triggerAreaCRDTEntity, transform, ref triggerAreaComponent);
+        }
+
+        private void ProcessOnEnterTriggerArea(in Entity triggerAreaEntity, in CRDTEntity triggerAreaCRDTEntity, in TransformComponent transform, ref SDKEntityTriggerAreaComponent triggerAreaComponent)
+        {
+            foreach (Collider entityCollider in triggerAreaComponent.EnteredEntitiesToBeProcessed)
+            {
+                PropagateResultComponent(triggerAreaEntity, triggerAreaCRDTEntity, transform.Transform,
+                    entityCollider, TriggerAreaEventType.TaetEnter, triggerAreaComponent.LayerMask);
+            }
+            triggerAreaComponent.TryClearEnteredAvatarsToBeProcessed();
+        }
+
+        private void ProcessOnStayInTriggerArea(in Entity triggerAreaEntity, in CRDTEntity triggerAreaCRDTEntity, in TransformComponent transform, ref SDKEntityTriggerAreaComponent triggerAreaComponent)
+        {
+            foreach (Collider entityCollider in triggerAreaComponent.CurrentEntitiesInside)
+            {
+                PropagateResultComponent(triggerAreaEntity, triggerAreaCRDTEntity, transform.Transform,
+                    entityCollider, TriggerAreaEventType.TaetStay, triggerAreaComponent.LayerMask);
+            }
+        }
+
+        private void ProcessOnExitTriggerArea(in Entity triggerAreaEntity, in CRDTEntity triggerAreaCRDTEntity, in TransformComponent transform, ref SDKEntityTriggerAreaComponent triggerAreaComponent)
+        {
+            foreach (Collider entityCollider in triggerAreaComponent.ExitedEntitiesToBeProcessed)
+            {
+                PropagateResultComponent(triggerAreaEntity, triggerAreaCRDTEntity, transform.Transform,
+                    entityCollider, TriggerAreaEventType.TaetExit, triggerAreaComponent.LayerMask);
+            }
+            triggerAreaComponent.TryClearExitedAvatarsToBeProcessed();
+        }
+
+        private void PropagateResultComponent(in Entity triggerAreaEntity, in CRDTEntity triggerAreaCRDTEntity, Transform triggerAreaTransform,
+            Collider triggerEntityCollider, TriggerAreaEventType eventType, ColliderLayer areaLayerMask)
+        {
+            Entity avatarEntity = Entity.Null;
+            ColliderSceneEntityInfo entityInfo = default;
+            if (triggerEntityCollider.gameObject.layer == PhysicsLayers.CHARACTER_LAYER
+                || triggerEntityCollider.gameObject.layer == PhysicsLayers.OTHER_AVATARS_LAYER)
+            {
+                if (!PhysicsLayers.LayerMaskContainsTargetLayer(areaLayerMask, ColliderLayer.ClPlayer)
+                    || !TryGetAvatarEntity(triggerEntityCollider.transform, out avatarEntity))
+                    return;
+            }
+
+            Decentraland.Common.Vector3 triggerEntityPos = default;
```

Comment:
Redundant initialization. Moreover, members are not declared as nullables, so they should not be assigned to `default` which is `null`

Reply:
tackled at https://github.com/decentraland/unity-explorer/pull/5244/commits/5368e4ebc06c7606f44abdd052c4cb2dcc6c1a56